### DISCUSSION TITLE
Access to an undefined property in json resource collection laravel

### DISCUSSION THREAD

**Discussion by waelsjaber** [larastan/larastan#1190] [Q&A] [answered]
I use json resource collection laravel

Currently I am getting:

```
Access to an undefined property App\Http\Resources\UserResource::$role
```
`$role` is pivot relationships.

```php
class UserResource extends JsonResource
{
    /**
     * Transform the resource into an array.
     *
     * @param  \Illuminate\Http\Request  $request
     * @return array
     */
    public function toArray($request)
    {
        return [
            'id' => $this->id,
            'role' => $this->role->name,
         ];
    }
}
```

is there a way to get this fixed?

**Comment by szepeviktor** [comment]
Please see our issue template on how to report a problem.
https://github.com/nunomaduro/larastan/issues/new

**Comment by szepeviktor** [comment, marked as answer]
The solution is already in our repo: https://github.com/nunomaduro/larastan/search?q=JsonResource
Please see all four files.

**Comment by canvural** [comment]
Hi,

You can add `@mixin User` to your class docblock to solve these.

**Reply by szepeviktor** [reply]
> You can add `@mixin User` to your class docblock to solve these.

Hello Can! Why do you give fish to users when they could have a net instead?

**Comment by waelsjaber** [comment]
@canvural 
I still face the issue and  used custom pivot accessor inside user model

```php
class User extends Authenticatable
{
    public function roles():BelongsToMany
    {
        return $this->belongsToMany(Role::class)
            ->as('role')
            ->withTimestamps();
    }
}    
```

**Reply by canvural** [reply]
Yes. Custom pivot accessors are not supported. And not possible to support. So only solution is to to add `@property Role $role` to your model docblock.
